src/main.py

```diff
@@ -495,75 +475,68 @@ def silence_listener(throwaway_frames):
 	inp.close()
 
 
-def loop():
-	global player
+trigger_thread = None
+def trigger_callback(trigger_name):
+	global trigger_thread
 
-	while True:
-		record_audio = False
+	triggers.disable()
+
+	trigger_thread = threading.Thread(target=trigger_process, args=(trigger_name,))
+	trigger_thread.setDaemon(True)
+	trigger_thread.start()
 
-		# Enable reading microphone raw data
-		inp = alsaaudio.PCM(alsaaudio.PCM_CAPTURE, alsaaudio.PCM_NORMAL, config['sound']['input_device'])
-		inp.setchannels(1)
-		inp.setrate(16000)
-		inp.setformat(alsaaudio.PCM_FORMAT_S16_LE)
-		inp.setperiodsize(1024)
 
-		while not record_audio:
-			time.sleep(.1)
+def trigger_process(trigger_name):
+	global player
+
+	trigger = triggers.triggers[trigger_name]
```

Comment:
So I've done that and then I thought - would it make more sense to go even further so that the trigger would pass its own instance? Like
`self._trigger_callback(self)`
and 
```
def trigger_callback(trigger):
	global trigger_thread

	triggers.disable()

	trigger_thread = threading.Thread(target=trigger_process, args=(trigger,))
	trigger_thread.setDaemon(True)
	trigger_thread.start()
```